expression/builtin_miscellaneous_vec.go

```diff
@@ -617,3 +618,38 @@ func (b *builtinReleaseLockSig) vecEvalInt(input *chunk.Chunk, result *chunk.Col
 	}
 	return nil
 }
+
+func (b *builtinVitessHashSig) vectorized() bool {
+	return true
+}
+
+func (b *builtinVitessHashSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	column, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(column)
+
+	result.ResizeInt64(n, true)
+	r64s := result.Uint64s()
+
+	if err := b.args[0].VecEvalInt(b.ctx, input, column); err != nil {
+		return err
+	}
+	for i := 0; i < n; i++ {
+		if column.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
+		var uintKey = column.GetUint64(i)
+		var hash uint64
+		if hash, err = vitess.HashUint64(uintKey); err != nil {
+			return err
+		}
+		result.SetNull(i, false)
+		r64s[i] = hash
+	}
```

Comment:
I do not think this would let me do less `IsNull` calls, as I would still need to check it for every row. Or am I missing something? See the commit with the change.